olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp

```diff
@@ -252,134 +381,244 @@ TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyFound) {
+      .WillByDefault(testing::WithoutArgs([]() {
+        return olp::http::SendOutcome(olp::http::ErrorCode::CANCELLED_ERROR);
+      }));
+
+  ApiLookupClient lookup_client(kHrn, settings_);
+  repository::CatalogRepository repository(kHrn, settings_, lookup_client);
+
+  std::promise<read::CatalogVersionResponse> promise;
+  auto future = promise.get_future();
+  repository.GetLatestVersion(request,
+                              [&](read::CatalogVersionResponse response) {
+                                promise.set_value(std::move(response));
+                              });
+
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  const auto result = future.get();
+
+  EXPECT_FALSE(result);
+  EXPECT_EQ(result.GetError().GetErrorCode(),
+            olp::client::ErrorCode::Cancelled);
+}
+
 TEST_F(CatalogRepositoryTest, GetLatestVersionCancelledBeforeExecution) {
   settings_.retry_settings.timeout = 0;
   olp::client::CancellationContext context;
 
-  auto request = read::CatalogVersionRequest();
+  const auto request = read::CatalogVersionRequest();
 
   ON_CALL(*network_, Send(_, _, _, _, _))
-      .WillByDefault([&](olp::http::NetworkRequest, olp::http::Network::Payload,
-                         olp::http::Network::Callback,
-                         olp::http::Network::HeaderCallback,
-                         olp::http::Network::DataCallback) {
+      .WillByDefault(testing::WithoutArgs([]() {
         ADD_FAILURE() << "Should not be called on cancelled operation";
         constexpr auto unused_request_id = 10;
         return olp::http::SendOutcome(unused_request_id);
-      });
+      }));
 
   context.CancelOperation();
 
   ApiLookupClient lookup_client(kHrn, settings_);
   repository::CatalogRepository repository(kHrn, settings_, lookup_client);
-  auto response = repository.GetLatestVersion(request, context);
+  const auto response = repository.GetLatestVersion(request, context);
 
-  ASSERT_FALSE(response.IsSuccessful());
+  ASSERT_FALSE(response);
   EXPECT_EQ(olp::client::ErrorCode::Cancelled,
             response.GetError().GetErrorCode());
 }
 
+TEST_F(CatalogRepositoryTest, AsyncGetLatestVersionCancelledBeforeExecution) {
+  settings_.retry_settings.timeout = 0;
+
+  const auto request = read::CatalogVersionRequest();
+
+  ON_CALL(*network_, Send(_, _, _, _, _))
+      .WillByDefault(testing::WithoutArgs([]() {
+        ADD_FAILURE() << "Should not be called on cancelled operation";
+        constexpr auto unused_request_id = 10;
+        return olp::http::SendOutcome(unused_request_id);
+      }));
+
+  ApiLookupClient lookup_client(kHrn, settings_);
+  repository::CatalogRepository repository(kHrn, settings_, lookup_client);
+
+  auto scheduler =
+      olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler();
+
+  std::promise<read::CatalogVersionResponse> promise;
+  auto future = promise.get_future();
+
+  auto cont =
+      olp::thread::TaskContinuation(std::move(scheduler))
```

Comment:
This appeared to be a unit test. In the unit tests, it is better to abstract from any dependencies to test a distinct class. So I wouldn't recommend using TaskContinuation here, as it may contain some issues under the hood. 
But it is still fine if this use-case cannot be tested at all without TaskContinuation.